app/config/Modules.scala

```diff
@@ -16,11 +16,15 @@
 
 package config
 
-import com.google.inject.AbstractModule
+import play.api.inject.Binding
+import play.api.{Configuration, Environment}
 
-class Modules extends AbstractModule {
+import java.time.Clock
 
-  override def configure(): Unit = {
-    bind(classOf[AppConfig]).to(classOf[AppConfig]).asEagerSingleton()
+class Modules extends play.api.inject.Module {
+  override def bindings(environment: Environment, configuration: Configuration): collection.Seq[Binding[_]] = {
+    Seq(
+      bind[AppConfig].to[AppConfigImpl].eagerly()
+    )
   }
 }
```

Comment:
This `Modules` class isn't actually referenced or enabled in `application.conf` - so, it may not be required. I'm guessing the binding at runtime atm is being provided purely by the `@ImplementBy` annotation.

So you could just delete this file potentially? Or add below to `application.conf`:
```
play.modules.enabled += "config.Modules"
```

Reply:
I _think_ that @implementedBy is supposed to do broadly the same thing as having a binding in a module but I will enable the module anyway for security because I'm not confident it won't cause issues somewhere to omit it